apollo-gradle-plugin/build.gradle.kts

```diff
@@ -1,114 +1,73 @@
-   * `extensions.findByName("com.apollographql.relocated.kotlin")
-   * See https://github.com/johnrengelman/shadow/issues/232 for more details
-   */
-  dependency.exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib")
-  dependency.exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib-jdk7")
-  dependency.exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib-jdk8")
-  dependency.exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib-common")
-  dependency.exclude(group = "org.jetbrains.kotlin", module = "kotlin-reflect")
+// Configuration for extra jar to pass to R8 to give it more context about what can be relocated
+configurations.create("gr8Classpath")
+// Configuration dependencies that will be shadowed
+val shadeConfiguration = configurations.create("shade")
 
-  dependencies.add("shadowImplementation", dependency)
-}
 dependencies {
   compileOnly(groovy.util.Eval.x(project, "x.dep.minGradleApi"))
   //compileOnly(groovy.util.Eval.x(project, "x.dep.gradleApi"))
   compileOnly(groovy.util.Eval.x(project, "x.dep.kotlinPluginMin"))
   compileOnly(groovy.util.Eval.x(project, "x.dep.android.minPlugin"))
 
-  addShadowImplementation(projects.apolloCompiler)
-  addShadowImplementation(projects.apolloAst)
-  addShadowImplementation(projects.apolloApi)
+  /**
+   * OkHttp has some bytecode that checks for Conscrypt at runtime (https://github.com/square/okhttp/blob/71427d373bfd449f80178792fe231f60e4c972db/okhttp/src/main/kotlin/okhttp3/internal/platform/ConscryptPlatform.kt#L59)
+   * Put this in the classpath so that R8 knows it can relocate DisabledHostnameVerifier as the superclass is not package-private
+   *
+   * Keep in sync with https://github.com/square/okhttp/blob/71427d373bfd449f80178792fe231f60e4c972db/buildSrc/src/main/kotlin/deps.kt#L24
+   */
+  add("gr8Classpath", "org.conscrypt:conscrypt-openjdk-uber:2.5.2")
+
+  add("shade", "org.jetbrains.kotlin:kotlin-stdlib")
+  add("shade", projects.apolloApi)
+  add("shade", projects.apolloCompiler)
+  add("shade", projects.apolloAst)
 
-  addShadowImplementation(create(groovy.util.Eval.x(project, "x.dep.okHttp.okHttp4")))
-  // Needed for manual Json construction in `SchemaDownloader`
-  addShadowImplementation(create(groovy.util.Eval.x(project, "x.dep.moshi.moshi")))
+  add("shade", groovy.util.Eval.x(project, "x.dep.okHttp.okHttp4"))
+  add("shade", groovy.util.Eval.x(project, "x.dep.moshi.moshi").toString()) {
+    because("Needed for manual Json construction in `SchemaDownloader`")
+  }
 
   testImplementation(groovy.util.Eval.x(project, "x.dep.junit"))
   testImplementation(groovy.util.Eval.x(project, "x.dep.truth"))
   testImplementation(groovy.util.Eval.x(project, "x.dep.assertj"))
   testImplementation(groovy.util.Eval.x(project, "x.dep.okHttp.mockWebServer4"))
 }
 
-val shadowJarTask = tasks.named("shadowJar", com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar::class.java)
-val shadowPrefix = "com.apollographql.relocated"
+if (true) {
```

Comment:
Is this to quickly disable it when working on the plugin?

Reply:
Yep, exactly!